src/app/product-details-page/components/product-description/product-image-preview/product-image-preview.component.spec.ts

```diff
@@ -9,19 +11,30 @@ describe('ProductImagePreviewComponent', () => {
 
   beforeEach(async(() => {
     TestBed.configureTestingModule({
-      declarations: [ ProductImagePreviewComponent, ZoomComponent ],
-      imports: [ RouterTestingModule ]
+      declarations: [ProductImagePreviewComponent, ProductDescriptionComponent, ProductOrderComponent, ZoomComponent],
+      imports: [RouterTestingModule]
     })
-    .compileComponents();
+      .compileComponents().then(() => {
+        fixture = TestBed.createComponent(ProductImagePreviewComponent);
+        component = fixture.componentInstance;
+        fixture.detectChanges();
+      });
   }));
 
-  beforeEach(() => {
-    fixture = TestBed.createComponent(ProductImagePreviewComponent);
-    component = fixture.componentInstance;
-    fixture.detectChanges();
-  });
-
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should render img if smallImagesArr has length', (() => {
+    const swatches = ['http://therxreview.com/wp-content/uploads/2013/05/Reebok-Track-Jacket.png',
+      'http://therxreview.com/wp-content/uploads/2013/05/Reebok-Track-Jacket.png',
+      'http://therxreview.com/wp-content/uploads/2013/05/Reebok-Track-Jacket.png',
+      'http://therxreview.com/wp-content/uploads/2013/05/Reebok-Track-Jacket.png'];
+    component.smallImagesArr = swatches;
```

Comment:
you didn't get my previous comment. In order to test the component, you should only change the input properties

component.swatches = swatches;

Reply:
done)